cranelift/codegen/src/machinst/sections.rs

```diff
@@ -74,6 +74,15 @@ impl MachSections {
         sink.end_codegen();
     }
 
+    /// Get a list of source location mapping tuples.
+    pub fn get_srclocs(&self) -> Vec<MachSrcLoc> {
```

Comment:
Yup, done; wrote a little custom iterator. (I might have been able to get it to work returning an `impl Iterator<Item = &'a MachSrcLoc>` but getting the types to align for the zero-sections case and otherwise is tricky, since impl-returns must match on all returning paths...)